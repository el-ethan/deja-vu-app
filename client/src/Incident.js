import React from 'react';

import ReactMarkdown from 'react-markdown';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import DeleteRoundedIcon from '@material-ui/icons/DeleteRounded';
import CreateIcon from '@material-ui/icons/Create';



function Incident({problem, solution, onDelete, _id}) {
    return (
        <Card className="incident-card">
            <DeleteRoundedIcon className="trash-can" onClick={() => onDelete(_id)}/>
            <CreateIcon className="pencil" />
            <CardContent>
                <div className="card-section">
    	            <Typography variant="h5" component="h2">Problem:</Typography>
                    <div className="stacktrace">
                        <ReactMarkdown id="problem-markdown" source={`\`\`\`\n${problem}\n\`\`\``} />
                    </div>
                </div>
                <div className="card-section">
                    <Typography variant="h5" component="h2">Solution:</Typography>
                    <ReactMarkdown id="solution-markdown" source={solution} />
                </div>
            </CardContent>
    	</Card>
    );
}

export default Incident;
