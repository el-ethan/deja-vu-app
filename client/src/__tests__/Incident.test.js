import React from 'react';
import { shallow } from 'enzyme';
import Incident from '../Incident';

describe('Inicdent', () => {
    it('should render correctly', () => {
        const component = shallow(<Incident problem="gremlins" solution="don't feed after midnight" />);
        expect(component).toMatchSnapshot();
    });
});
