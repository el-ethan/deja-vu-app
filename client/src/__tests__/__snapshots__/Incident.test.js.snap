// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inicdent should render correctly 1`] = `
<WithStyles(Card)
  className="incident-card"
>
  <pure(DeleteRoundedIcon)
    className="trash-can"
    onClick={[Function]}
  />
  <pure(CreateIcon)
    className="pencil"
  />
  <WithStyles(CardContent)>
    <div
      className="card-section"
    >
      <WithStyles(Typography)
        component="h2"
        variant="h5"
      >
        Problem:
      </WithStyles(Typography)>
      <div
        className="stacktrace"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          id="problem-markdown"
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="\`\`\`
gremlins
\`\`\`"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
    </div>
    <div
      className="card-section"
    >
      <WithStyles(Typography)
        component="h2"
        variant="h5"
      >
        Solution:
      </WithStyles(Typography)>
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        id="solution-markdown"
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        source="don't feed after midnight"
        sourcePos={false}
        transformLinkUri={[Function]}
      />
    </div>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
